mitmproxy/addons/view.py

```diff
@@ -68,32 +71,49 @@ def __call__(self, f):
 
 
 class OrderRequestStart(_OrderKey):
-    def generate(self, f: http.HTTPFlow) -> int:
-        return f.request.timestamp_start or 0
+    def generate(self, f: mitmproxy.flow.Flow) -> float:
+        return f.timestamp_start
 
 
 class OrderRequestMethod(_OrderKey):
-    def generate(self, f: http.HTTPFlow) -> str:
-        return f.request.method
+    def generate(self, f: mitmproxy.flow.Flow) -> str:
+        if isinstance(f, http.HTTPFlow):
+            return f.request.method
+        elif isinstance(f, tcp.TCPFlow):
+            return "TCP"
+        else:
+            raise NotImplementedError()
 
 
 class OrderRequestURL(_OrderKey):
-    def generate(self, f: http.HTTPFlow) -> str:
-        return f.request.url
+    def generate(self, f: mitmproxy.flow.Flow) -> str:
+        if isinstance(f, http.HTTPFlow):
+            return f.request.url
+        elif isinstance(f, tcp.TCPFlow):
+            return human.format_address(f.server_conn.address)
+        else:
+            raise NotImplementedError()
 
 
 class OrderKeySize(_OrderKey):
-    def generate(self, f: http.HTTPFlow) -> int:
-        s = 0
-        if f.request.raw_content:
-            s += len(f.request.raw_content)
-        if f.response and f.response.raw_content:
-            s += len(f.response.raw_content)
-        return s
-
+    def generate(self, f: mitmproxy.flow.Flow) -> int:
+        if isinstance(f, http.HTTPFlow):
+            size = 0
+            if f.request.raw_content:
```

Comment:
In `clientplayback.py` you're using `if f.request.raw_content is None:`. Should this be consistent or is there a difference between the two? I'm just looking through this PR as I'm learning Python.

Reply:
Good find, but it doesn't really matter here. :-)

.raw_content is of type Optional[bytes], which only evaluates to False if it is `None` or `b''`. We need to check for `None` because `len(None)` would rightly error, but it doesn't matter for `b''` as that would just add zero to size. I usually go for the shortest form in these cases, even though `is not None` would work just as well here.

Reply:
Gotcha, so it's like `foo === undefined` vs relying on falsy values (`!foo`) in JavaScript and this catches both None and the empty value. I guess it's a matter of taste, I think I'm spoiled by using Go where everything is pretty explicit and self-explanatory (I mean Python probably beats JavaScript in this regard :smile: )